### ISSUE TITLE
Praat generates 0 KB PNG file for audio longer than 15 min

### ISSUE THREAD

**Issue by fahadash** [praat/praat#189]
Script is attached with this issue. We can share the audio that is causing the problem if need be. We ran the script through command line as well as the UI, we did see the Praat Picture fine, but the Save as PNG File... instruction is creating 0 bytes file. Any help on this will be appreciated.

You may have to correct the paths in the attached script
[waveform.praat.txt](https://github.com/praat/praat/files/151985/waveform.praat.txt)

**Comment by fahadash** [comment]
Hello, I would love to get an update on this one. I honestly have the least grip on the project or else I would've looked into it myself.

**Comment by jjatria** [comment]
I cannot reproduce this on my machine (Kubuntu 15.10 32-bit with Praat 6.0.16).

What version of Praat are you using? What operating system?

This command seems to have been added in [5.4](https://github.com/praat/praat/releases/tag/v5.4), and there were bug fixes for it in [5.4.13](https://github.com/praat/praat/releases/tag/v5.4.13) and [5.4.22](https://github.com/praat/praat/releases/tag/v5.4.22). 

The relevant pieces of code [appear to be](https://github.com/praat/praat/search?utf8=%E2%9C%93&q=Picture_writeToPngFile_300) in [Picture.h](https://github.com/praat/praat/blob/e63090f/sys/Picture.h#L115-L116), [Picture.cpp](https://github.com/praat/praat/blob/8892671/sys/Picture.cpp#L435-L451), and specially [praat_picture.cpp](https://github.com/praat/praat/blob/8892671/sys/praat_picture.cpp#L517-L563).

**Comment by fahadash** [comment]
@jjatria Thank you for your response. Here is the output of --version
PS C:\praat> .\Praat.exe --version
PS C:\praat> Praat 6.0.14 (February 11 2016)

I am using Windows Server 2012 R2.
I also tried Windows 8.1 Professional

**Comment by jjatria** [comment]
I don't know if I can be much help with Windows, really.

Does the problem _not_ exist with sounds _shorter_ than 15 minutes?

**Comment by fahadash** [comment]
@jjatria The 15 minute is not exact, we have noticed this done on slightly shorter audio files too. But yes we never had issues with files shorter than 10 minutes. I honestly don't have the build environment set up and not the full technical expertise to try to figure it out myself. Any pointers would be appreciated.

**Comment by fahadash** [comment]
@jjatria Thank you for your help. Is there any way we could escalate this? I could provide you the audio file I am using, I may give you the access to the AWS server where I have praat set up.

Thanks,
fahadash

**Comment by shrestha2lt8** [comment]
Hi Guys,

Is there anyone who figure it out this issue ?

**Comment by PaulBoersma** [comment]
Your script contains the line
`view_width = 4 * total_duration/15`
later followed by
`Viewport... 0 view_width 0 1`
which means that for a sound 15 minutes long you are creating a PNG picture that is 240 inches wide.
My hypothesis is that on Windows, a PNG file somehow cannot be wider than 2^32 = 65536 pixels, which is 65536/300 = 218 inches, or 218*15/4 = 819.2 seconds, or 13 minutes and 39.2 seconds. This is despite the fact that the width information in a PNG file is four bytes, not two, and so is the width information in the bitmap that Praat draws into. By the way, was it really your intent to create pictures that are 6 meters wide?

**Comment by fahadash** [comment]
@PaulBoersma yes this is intentional. If we could get a different format that works on web, we are okay.

**Comment by jjatria** [comment]
Is tiling a possibility? 

If shorter files work, you could extract smaller parts of the sound file and generate PNG files for those, and then tile them together using a graphics library like they do with maps. I know [Leaflet](http://leafletjs.com/) has features for [tiling with your own files](http://build-failed.blogspot.co.uk/2012/11/zoomable-image-with-leaflet.html), even those that are not map-related.

**Comment by fahadash** [comment]
@jjatria That is what we have decided to do if we find out that we could not get an image file that long at all.

**Comment by PaulBoersma** [comment]
The cause of the restriction may be that Praat still has to use the 1995 Windows API for drawing graphics ("GDI"). Once we don't have to support Windows XP any longer, we will be able to use Direct2D instead. If I have anything better to say before that time, I'll get back about it here. Anyway, this is the first time I heard that people need such wide PNG pictures, and it's good to hear that such pictures can be useful.

**Comment by drammock** [comment]
> The cause of the restriction may be that Praat still has to use the 1995 Windows API for drawing graphics ("GDI"). Once we don't have to support Windows XP any longer, we will be able to use Direct2D instead.

Microsoft stopped supporting windows XP in 2014.  When will praat drop support for XP, enabling the switch from GDI to Direct2D @PaulBoersma?
